server/src/models/map.ts

```diff
@@ -0,0 +1,46 @@
+import { Table, Model, Column, DataType, PrimaryKey, AutoIncrement, AllowNull, HasMany } from "sequelize-typescript"
+
+import { Sandtrap } from "./sandtrap"
+import { Water } from "./water"
+import { Barrier } from "./barrier"
+
+@Table({ tableName: "Maps" })
+export class Map extends Model<Map> {
+  @PrimaryKey
+  @AutoIncrement
+  @Column(DataType.INTEGER)
+  id!: number
+
+  @AllowNull(false)
+  @Column(DataType.DATEONLY)
+  day!: Date
```

Comment:
Since we're consistently doing lookups on the date column, indexing it for quicker lookups would be ideal.